### ISSUE TITLE
Mapbox Style with multiple sources not being rendered correctly

### ISSUE THREAD

**Issue by cchandel** [iTowns/itowns#1927]
Hi,
    I'm trying to use a Mapbox vector style that is working fine for a 2D map using Openalyers + ol-maplibre-layer.
    Using the same json file as style leads to only 1 layer being shown.

## Your Environment

* Version used: 2.38.2
* Browser Name and version: chrome 106.0.5249.119
* Operating System and version (desktop or mobile): Linux Desktop
* Link to your project: 

## Context

  The style json file has multiple sources as below :-
```
  "sources": {
    "physical": {
      "type": "vector",
      "tiles": [
        "http://localhost:8000/services/physical/tiles/{z}/{x}/{y}.pbf"
      ],
      "minZoom": 2,
      "maxZoom": 14
    },
    "land": {
      "type": "vector",
      "tiles": [
        "http://localhost:8000/services/land/tiles/{z}/{x}/{y}.pbf"
      ],
      "minZoom": 2,
      "maxZoom": 14
    }
  }
  "layers":[
    { 
      {
      "id": "marine",
      "type": "fill",
      "source": "physical",
      "source-layer": "marine",
     },
   },
   {
      "id": "land",
      "type": "fill",
      "source": "land",
      "source-layer": "land",
      "paint": {
        "fill-color": "#ccffff",
        "fill-antialias": true
      }
    },
 ]
```

## Steps to Reproduce (for bugs)

1.  Use a mapbox style with multiple sources 

## Expected Behavior

All the layers referencing all the sources should be displayed.

## Actual Behavior

Only the first layer referencing the first source is displayed

All the layers referencing all the sources should be displayed.

**Comment by mgermerie** [comment]
Hi, and thanks for this feedback !

Indeed, iTowns `VectorTileSource` only considers the first `source` specified in Mapbox vector style. This is partly due to [this piece of code](https://github.com/iTowns/itowns/blob/9ee991c3cd6595de6c058cd2cf81976a3daab7e0/src/Source/VectorTilesSource.js#L90). 

The issue is that `VectorTileSource` has a single `url` property, therefore referencing a single `source` and not an array of `sources` for example. Implementing the support for Mapbox style with multiple sources would require some change on this, and also on fetching and parsing methods used for Mapbox data.

@PhilippeMarc-IGN and @LoicUTC, maybe we could add this feature to our next releases plannings ?

**Comment by a-dutremble** [comment]
Hi ! I am currently working on this issue.

Can you send me an example of data that works ? If the data is private, you can send it to me by mail if you want at aymeric.dutremble@gmail.com.

**Comment by ftoromanoff** [comment]
@cchandel Hello, we just merged PR #2128 which should allow the use of multi source on VectorTileSource.
Can you test with your data on itowns@next ?
